compare_solver_exercise.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## ------- import packages -------
+import networkx as nx
+import dimod
+# TODO:  Add code here to import your sampler
+
+# TODO:  Add code here to import your Traveling Salesman QUBO generator
+
+
+## ------- provided distance matrix -------
+distance_matrix = [
+        [0, 2230, 1631, 1566, 1346, 1352, 1204],
+        [2230, 0, 845, 707, 1001, 947, 1484],
+        [1631, 845, 0, 627, 773, 424, 644],
+        [1566, 707, 627, 0, 302, 341, 1027],
+        [1346, 1001, 773, 302, 0, 368, 916],
+        [1352, 947, 424, 341, 368, 0, 702],
+        [1204, 1484, 644, 1027, 916, 702, 0],
+    ]
+
+
+# TODO:  Add code here to define your QUBO dictionary
+def get_qubo(G):
+    """Returns a dictionary representing a QUBO"""
+
+    # Add QUBO construction here
+
+    return Q
+
+
+# TODO:  Add code here to define your sampler
+def get_sampler():
+    """Returns a sampler"""
+
+    # Add sampler here
+
+    return sampler
+
+
+## ------- Main program -------
+if __name__ == "__main__":
+
+    G = nx.Graph()
+    j = 0
+    for row in distance_matrix:
+        for k, item in enumerate(row):
+            if j < k:
+                G.add_edge(j, k, weight=item)
+        j += 1
+
+    Q = get_qubo(G)
+    sampler = get_sampler()
+    bqm = dimod.BinaryQuadraticModel.from_qubo(Q)
+    response = sampler.sample(bqm)
```

Comment:
if you're going to use QUBO, why go through the BQM at all? Can just do
```
response = sampler.sample_qubo(Q)
```

Reply:
Yes, that's certainly true. I wanted to leave open the possibility of using an offset. But if you're feeling it would be better, I can make that change.

Reply:
Could we add in the offset? It's mentioned in the lecture so I think it would be nice to bring in here too.

Reply:
I didn't do it because it isn't produced by get_qubo. I'm happy to have them think about it and compute it - what do you think?

Reply:
If you want the students to add it in you may need to add some comments to help them with that in the lecture video.

Reply:
I added it, to the get_qubo method. Please have a look.